### PR TITLE
P1sar/feat/deployment guide and tf fixes

### PR DESCRIPTION
**Pull request by P1sar**
Related to https://github.com/sygmaprotocol/project-backlog/issues/60

### CODE REVIEW

**Review by eedygreen — APPROVED** [review]

File: .github/workflows/deploy_ecs_TESTNET.yml [hunk 1]
```diff
@@ -11,14 +11,15 @@ env:
...
+  AWS_ROLE: '${{ secrets.AWS_ROLE_NAME }}'
 
 jobs:
   deploy:
     name: deploy
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        relayer_id: [0, 1, 2]
+        relayer_id: [0]
```
Comment (mpetrun5) [inline .github/workflows/deploy_ecs_TESTNET.yml:22]: Why was this changed?

**Review by mpetrun5 — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy_ecs_TESTNET.yml (hunk 1 above)
Comment (P1sar) [reply .github/workflows/deploy_ecs_TESTNET.yml:22]: By default I want this script to deploy only one relayer and add description how to increase amount of relayers later.

Comment (mpetrun5) [reply .github/workflows/deploy_ecs_TESTNET.yml:22]: Just realized I was looking at the wrong repo and thought these were our current testnet task files.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,25 @@
-# Sygma relayer Deployment
+# Sygma relayer Deployment guide
+This guide is a ste-by-stp manual that will help you deploy Sygma Relayer and join Sygma Bridge MPC consensus.
```
Comment (mpetrun5) [inline README.md]:
```suggestion
This guide is a step-by-step manual that will help you deploy Sygma Relayer and join Sygma Bridge MPC consensus.
```

**Review by mpetrun5 — CHANGES_REQUESTED** [review]
Can you please run it through a spell checker there are a lot of small spelling errors?

File: README.md (hunk 2 above)
Comment (P1sar) [reply README.md]: Fixed spell errors

File: README.md [hunk 3]
```diff
@@ -44,18 +78,137 @@ You might need to input a few variables, you can do that in the prompt that will
...
+#### Set Secret variables
+Set the next secret variables in GitHub Secret variables section. They will be used for deployment.
+
+ `AWS_REGION` - a region where you want your relayer to be deployed
+ `AWS_ARN` - ARN of the user that will be deploying the Relayer
+ `AWS_ROLE` - the name of IAM Role that will perform deployment
+
+#### Cofigurate other GitHub action variables.
+
+You can also configure different [env vatiables](https://github.com/sygmaprotocol/sygma-relayer-deployment/blob/main/.github/workflows/deploy_ecs_TESTNET.yml#L8)
```
Comment (mpetrun5) [inline README.md]:
```suggestion
You can also configure different [env variables](https://github.com/sygmaprotocol/sygma-relayer-deployment/blob/main/.github/workflows/deploy_ecs_TESTNET.yml#L8)
```

File: README.md [hunk 4]
```diff
@@ -44,18 +78,137 @@ You might need to input a few variables, you can do that in the prompt that will
...
+        "fresh": true,
+    }
+
+    {
+        "id": 0,
+        "name": "mumbai",
+        "type": "evm", // types {evm, substrate}
+        "key": "123", // Private key that used to send execution transactions
+        "endpoint": "{RPC endpoint}",
+        "startBlock": 8415289, // the block from where you Relayer should start. (In mos times it should be lastest block)
```
Comment (mpetrun5) [inline README.md]:
```suggestion
        "startBlock": 8415289, // the block from where you Relayer should start. (Most times it should be lastest block)
```

File: README.md [hunk 5]
```diff
@@ -44,18 +78,137 @@ You might need to input a few variables, you can do that in the prompt that will
...
+        "fresh": true,
+    }
+
+    {
+        "id": 0,
+        "name": "moonbase",
+        "type": "evm", // types {evm, substrate}
+        "key": "123", // Private key that used to send execution transactions
+        "endpoint": "{RPC endpoint}",
+        "startBlock": 8415289, // the block from where your Relayer should start. (In most times it should be lastest block)
```
Comment (mpetrun5) [inline README.md]:
```suggestion
        "startBlock": 8415289, // the block from where your Relayer should start. (Most times it should be lastest block)
```

File: README.md [hunk 6]
```diff
@@ -44,18 +78,137 @@ You might need to input a few variables, you can do that in the prompt that will
...
+- **SYG_RELAYER_MPCCONFIG_TOPOLOGYCONFIGURATION_URL**
+
+    URL to fetch topology file from.  The Sygma team will provide you with this key.
+
+- **SYG_RELAYER_MPCCONFIG_KEYSHAREPATH**
+
+    Example: `/mount/r1.keyshare` - path to the file that will contain MPC private key share. Should be unique per relayer. (eg /mount/r1.keyshare, /mount/r2.keyshare, /mount/r3.keyshare, etc)
+
+
+### Lunch relayer
```
Comment (mpetrun5) [inline README.md]:
```suggestion
### Launching a relayer
```

File: README.md [hunk 7]
```diff
@@ -44,18 +78,137 @@ You might need to input a few variables, you can do that in the prompt that will
...
+
+After all, the configuration parameters above are set we need to add your Relayer(s) to the Sygma MPC network by updating the network topology.
```
Comment (mpetrun5) [inline README.md]:
```suggestion
After all the configuration parameters above are set we need to add your Relayer(s) to the Sygma MPC network by updating the network topology.
```

File: README.md [hunk 8]
```diff
@@ -44,18 +78,137 @@ You might need to input a few variables, you can do that in the prompt that will
...
+
+### Run the deployment
+After configuration is done, the current pipeline will run on every **push to the main branch**.
+Make sure to grant all necessary permissions to the pipeline, either using IAM Roles or Environment Variables.
+
+
+### Confirm
+Go to the logs.
+You should see something like
+`Processing 0 deposit events in blocks: 3422205 to 3422209` - this is the log that parses the blocks in search of bridging events. That means that relayer up and listening.
```
Comment (mpetrun5) [inline README.md]:
```suggestion
`Processing 0 deposit events in blocks: 3422205 to 3422209` - this is the log that parses the blocks in search of bridging events. That means that relayer is up and listening.
```

File: README.md [hunk 9]
```diff
@@ -44,18 +78,137 @@ You might need to input a few variables, you can do that in the prompt that will
...
+
+
+## Other
+
+#### To change the number of relayers to deploy
+Amount of IDS in array set amount of relayers to be deployed.
+
+#### Relayer shared configuration
+Relayer configuration is done with `--config-url` flag on Relayer start and can be changed [here](https://github.com/sygmaprotocol/sygma-relayer-deployment/blob/main/ecs/task_definition_TESTNET.j2#L23)
+This flag set-up shared configuration IPNS URL that is used by all Relayers in the MPC network and provided by Sygma.
```
Comment (mpetrun5) [inline README.md]:
```suggestion
This flag sets up shared configuration IPNS URL that is used by all Relayers in the MPC network and provided by Sygma.
```

**Review by mpetrun5 — APPROVED** [review]
